pkg/operator/configobservation/apiserver/observe_cors.go

```diff
@@ -0,0 +1,28 @@
+package apiserver
+
+import (
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// ObserveDefaultUserServingCertificate returns an ObserveConfigFunc that observes user configuration
+// of CORSAllowedOrigins
+var ObserveExtraCORSAllowedOrigins configobserver.ObserveConfigFunc = (&apiServerObserver{
```

Comment:
please don't re-use apiServerObserver .  I think that pattern is overkill for your needs.  try to model it on an easier one like `ObserveInternalRegistryHostname`